src/crimson/os/seastore/transaction_manager.h

```diff
@@ -860,18 +860,21 @@ class TransactionManager : public ExtentCallbackInterface {
       pref.is_indirect() ?
 	pref.get_intermediate_length() :
 	pref.get_length(),
-      [pin=std::move(pin)]
+      [&pref]
       (T &extent) mutable {
 	assert(!extent.has_laddr());
 	assert(!extent.has_been_invalidated());
-	assert(!pin->has_been_invalidated());
-	assert(pin->get_parent());
-	pin->link_child(&extent);
-	extent.maybe_set_intermediate_laddr(*pin);
+	assert(!pref.has_been_invalidated());
+	assert(pref.get_parent());
+	pref.link_child(&extent);
+	extent.maybe_set_intermediate_laddr(pref);
       }
-    ).si_then([FNAME, &t](auto ref) mutable -> ret {
+    ).si_then([FNAME, &t, pin=std::move(pin)](auto ref) mutable -> ret {
       SUBTRACET(seastore_tm, "got extent -- {}", t, *ref);
       assert(ref->is_fully_loaded());
+      ceph_assert(pin->get_checksum() == 0 || // TODO: remapped extents may
+					      // not have recorded chksums
```

Comment:
IIUC this comment isn't accurate yet.

I think we should firstly ignore checking the crc explicitly for the indirect pin, e.g.

```c++
if (!pin->is_indirect()) {
  ceph_assert(pin->get_checksum() == 0 || // TODO: remapped extents may
			                  // not have recorded chksums
  pin->get_checksum() == ref->get_crc32c());
}
```

Reply:
> IIUC this comment isn't accurate yet.
> 
> I think we should firstly ignore checking the crc explicitly for the indirect pin, e.g.
> 
> ```c++
> if (!pin->is_indirect()) {
>   ceph_assert(pin->get_checksum() == 0 || // TODO: remapped extents may
> 			                  // not have recorded chksums
>   pin->get_checksum() == ref->get_crc32c());
> }
> ```

I think indirect mapping's checksum is also 0, am I right? If so, we don't need to check `!pin->is_indirect()`, right?

Reply:
How about this?
```c++
if (pin->is_indirect()) {
  ceph_assert(pin->get_checksum() == 0);
} else {
  ceph_assert(pin->get_checksum() == 0 || // TODO: remapped extents may
			                  // not have recorded chksums
              pin->get_checksum() == ref->get_crc32c());
}
```
I think this more accurately claimed that indirect pin's checksum must be 0.